<launch>
  <machine address="localhost" name="local" />
  <arg name="vision_computer" default="local" />
  <arg name="robot" default="puddles" />

  <include file="$(find riptide_hardware)/launch/computer_monitor.launch" />
  <include file="$(find riptide_hardware)/launch/computer_monitor.launch" unless="$(eval vision_computer == 'local')">
    <arg name="computer" value="$(arg vision_computer)" />
  </include>

  <node pkg="riptide_hardware" type="electrical_monitor.py" name="electrical_diags" output="screen">
    <param name="diag_thresholds_file" type="string" value="$(find riptide_hardware)/cfg/diagnostic_thresholds.yaml" />
    <param name="current_robot" value="$(arg robot)" />
  </node>
  <node pkg="riptide_hardware" type="voltage_monitor.py" name="voltage_diags" output="screen">
    <param name="diag_thresholds_file" type="string" value="$(find riptide_hardware)/cfg/diagnostic_thresholds.yaml" />
    <param name="current_robot" value="$(arg robot)" />
  </node>
  <node pkg="riptide_hardware" type="sensor_monitor.py" name="sensor_diags" output="screen" />

  <node pkg="diagnostic_aggregator" type="aggregator_node"
        name="diagnostic_aggregator" >
    <!-- Load the file you made above -->
    <rosparam command="load" 
              file="$(find riptide_hardware)/cfg/diagnostic_analyzers.yaml" />
  </node>
</launch>
